src/irmin/tree.ml

```diff
@@ -774,32 +774,54 @@ module Make (P : Private.S) = struct
         type acc.
         force:acc force ->
         uniq:uniq ->
-        pre:acc node_fn ->
-        post:acc node_fn ->
+        ?pre:acc node_fn ->
+        ?post:acc node_fn ->
```

Comment:
```suggestion
        pre:acc node_fn option ->
        post:acc node_fn option ->
```

(Let's keep things explicit unless there's value otherwise.)